oop_Session1/DOM.java

```diff
@@ -0,0 +1,277 @@
+			if(return_tag != null){
+				
+				// returns return_tag
+				return return_tag;
+			}
+				
+		}
+		// if element not found returns null
+		return null;
+	}
+	
+	/**
+	 * 
+	 * Searches element by ID in Composite element list
+	 * @param tag is the element object that has to checked for the passed id
+	 * @param id contains the id of the tag that has to be returned
+	 * @return Element object containing the tag of the id that was passed
+	 *
+	 */
+	Element checkIdInComposite(Element tag, String id){
+		
+		// loop to check id of element in list of composite element
+		for(Element tagInList: ((CompositeElement) tag).tagList()) {
+			
+			// calling function for checking id
+			Element return_tag = returnElementById(tagInList , id);
+			
+			// checking return_tag if null or not
+			if(return_tag != null){
+				
+				// return tag
+				return return_tag;
+			}
+	
+		}
+		
+		// if element not found returns null
+		return  null;
+	}
+	
+	/**
+	 * 
+	 * method that checks the passed tag's id with the id passed as parameter
+	 * @param tag is the element object that has to checked for the passed id
+	 * @param id contains the id of the tag that has to be returned
+	 * @return Element object containing the tag of the id that was passed
+	 */
+	private Element returnElementById(Element tag, String id) {
+		
+		// checks tag id
+		if ( id == tag.getId()) {
+			
+			// return tag
+			return tag;
+		}
+	
+		// check if tagInList is a composite element or not
+		if(tag.hasInstance()) {
+			
+			// recursive call for checking list of tagInList element
+			Element returnTag = checkIdInComposite(tag, id);
+			
+			// checks if return_tag is null or not
+			if (returnTag != null){
+				
+				// returns tag
+				return returnTag;
+			}
+		}
+		
+		// if element not found then return null
+		return null;
+	}
+
+	/**
+	 * Searches element by class name
+	 * @param className contains name of class whose tag has to be searched
+	 * @returns Element's List-list of elements having same class as passed
+	 * 
+	 */
+	List<Element> findElementByClass(String className) {
+		
+		// list contains element of same class
+		List<Element> list = new ArrayList<Element>();
+		
+		// loop for checking DOM element class name
+		for( Element tag : element) {
+			
+			// add elements to return list
+			list.addAll(AddElementByClass(tag , className));
+		}
+		
+		// returns list
+		return list;
+	}
+	
+	/**
+	 *
+	 * checks if the classname of the tag is same as parameter classname 
+	 * then adds in the list
+	 * @param tag contains the tag name whose classname will be compared with 
+	 *        passed classname
+	 * @param className contains name of class whose tag has to be searched
+	 * @return list containing tags
+	 */
+	private List<Element> AddElementByClass(Element tag,
+			String className) {
+		
+		// List contains tags
+		List<Element> list = new ArrayList<Element>();
+		
+		// checks tag's class name
+		if ( className == ((Element) tag).getClassName()) {

```

Comment:
**`String` instances can not be compared using `==` !!**; they should be compared with `.equals()`